packages/teams-js/test/public/people.spec.ts

```diff
@@ -26,270 +26,233 @@ describe('peoplePicker', () => {
       app._uninitialize();
     }
   });
+  const allowedContexts = [FrameContexts.content, FrameContexts.task, FrameContexts.settings];
+  const result = [
+    {
+      objectId: '5842943a-aa5a-470a-bfdc-7311b9988962',
+      displayName: 'Sonal Jha',
+      email: 'sojh@m365x347208.onmicrosoft.com',
+    } as people.PeoplePickerResult,
+  ];
+  const input = {
+    title: 'Title',
+    openOrgWideSearchInChatOrChannel: true,
+  };
+  describe('peoplePicker', () => {
+    /**
+     * People Picker tests
+     */
+    it('should not allow selectPeople calls before initialization', async () => {
+      expect(() => people.selectPeople()).toThrowError('The library has not yet been initialized');
+    });
+    Object.values(FrameContexts).forEach(context => {
+      if (allowedContexts.some(allowedContext => allowedContext === context)) {
+        return;
+      }
+
+      it(`should not allow selectPeople calls from the wrong context. context: ${context}`, async () => {
+        await mobilePlatformMock.initializeWithContext(context);
+
+        expect(() => people.selectPeople()).toThrowError(
+          `This call is only allowed in following contexts: ${JSON.stringify(
+            allowedContexts,
+          )}. Current context: "${context}".`,
+        );
+      });
+    });
 
-  /**
-   * People Picker tests
-   */
-  it('should allow selectPeople calls with null peoplePickerInputs', async () => {
-    await mobilePlatformMock.initializeWithContext(FrameContexts.task);
-    mobilePlatformMock.setClientSupportedSDKVersion(minVersionForSelectPeople);
-    return expect(people.selectPeople(null)).resolves;
-  });
-
-  it('should allow selectPeople calls with no peoplePickerInputs', async () => {
-    await mobilePlatformMock.initializeWithContext(FrameContexts.task);
-    mobilePlatformMock.setClientSupportedSDKVersion(minVersionForSelectPeople);
-    return expect(people.selectPeople()).resolves;
-  });
-
-  it('should allow selectPeople calls with undefined peoplePickerInputs', async () => {
-    await mobilePlatformMock.initializeWithContext(FrameContexts.task);
-    mobilePlatformMock.setClientSupportedSDKVersion(minVersionForSelectPeople);
-    return expect(people.selectPeople(undefined)).resolves;
-  });
+    Object.values(FrameContexts).forEach(context => {
+      if (!allowedContexts.some(allowedContext => allowedContext === context)) {
+        return;
```

Comment:
Similarly, avoid using return and put test cases in the required if and else block.